lib/config/configuration.go

```diff
@@ -1514,6 +1515,17 @@ func applyDatabasesConfig(fc *FileConfig, cfg *servicecfg.Config) error {
 	return nil
 }
 
+func convOracleOptions(o *DatabaseOracle) *servicecfg.OracleOptions {
+	if o == nil {
+		return nil
+	}
```

Comment:
You'd probably avoid a bunch of these `if o == nil` checks if you made it non-nullable in the proto spec.